File: 04 Kontrollstrukturen/Pascal/Aufgabe9.pas
Program Aufgabe9;

{$codepage utf8}

Var Zahl1: Integer;
    Zahl2: Integer;
    I: Integer;

Begin
    Write('Bitte gib die erste Zahl ein: ');
    Read(Zahl1);

    Write('Bitte gib die zweite Zahl ein: ');
    Read(Zahl2);

    If Zahl1 > Zahl2 Then // Zahlen tauschen
    Begin
        Zahl1 := Zahl1 xor Zahl2;
        Zahl2 := Zahl2 xor Zahl1;
        Zahl1 := Zahl1 xor Zahl2;
    End;

    WriteLn('Alle durch 7 teilbare Zahlen zwischen ', Zahl1, ' und ', Zahl2, ':');

    For I := Zahl1 To Zahl2 Do
        If I mod 7 = 0 Then
            WriteLn(I);
End.

File: 04 Kontrollstrukturen/Go/Aufgabe9.go
package main

import "fmt"

var (
	zahl1 int
	zahl2 int
	i int
)

func main() {
	fmt.Print("Bitte gib die erste Zahl ein: ")
	fmt.Scan(&zahl1)

	fmt.Print("Bitte gib die zweite Zahl ein: ")
	fmt.Scan(&zahl2)

	if zahl1 > zahl2 { // Zahlen tauschen
		zahl1 ^= zahl2
		zahl2 ^= zahl1
		zahl1 ^= zahl2
	}

	fmt.Println("Alle durch 7 teilbaren Zahlen zwischen", zahl1, "und", zahl2, ":")

	for i = zahl1; i < zahl2 + 1; i++ {
		if i % 7 == 0 {
			fmt.Println(i)
		}
	}
}